IndomitableIbises/Bangazon-API-Sprint-1
# Employee GET/POST/UPDATE/DELETE

## Link to Ticket
Closes #6

## Description of Proposed Changes

- Employee Model, View, URL, Serializer created so a user can manually create an Employee and get, post, update, delete and assign as supervisor.

- Within Model.py, View.py, Url.py and Serializer.py, I added a separation comment to keep Employee and Customer related classes apart for easier code navigation.

## Steps to Test

Outline the steps to test

```
git fetch --all
git checkout RC-Employee
```
1. code .
2. Delete existing db.sqlite3 file
3. Run ./django_data.sh API faker_factory for **PC**
or
django_data.sh API faker_factory for **MAC**
4.  Look over Employee Model, View, Url, Serializer -- let me know if you have any questions
5. python manage.py runserver
6. You should see three clickable urls on our page. Click on "http://localhost:8000/employee/" and make sure you are able to add an employee and assign as a supervisor as well as update an employee to unassign as supervisor.  Note:  foreign keys are not enabled on this table as of now.


## Impacted Areas in Application

List general components of the application that this PR will affect:

*models.py
*serializers.py
*urls.py
*views.py

## Mentions @username
@CashewRose @jcarter0149 @sirwin4 

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
   1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
   1. Every class must be documented with purpose, author, and methods.
   1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each
